File: python/courtlistener/api/retention_events.py
"""
Retention Events API module for CourtListener SDK.
"""

from typing import Dict, Any, Optional, List
from .base import BaseAPI


class RetentionEventsAPI(BaseAPI):
    """API for accessing data retention events."""
    
    def __init__(self, client):
        super().__init__(client)
        self.endpoint = "retention-events/"
    
    def list(self, **kwargs) -> Dict[str, Any]:
        """
        List retention events.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            API response with paginated list of retention events
        """
        return self.client.get(self.endpoint, params=kwargs)
    
    def get(self, event_id: int) -> Dict[str, Any]:
        """
        Get a specific retention event by ID.
        
        Args:
            event_id: Retention event ID
        
        Returns:
            Retention event data
        """
        return self.client.get(f"{self.endpoint}{event_id}/")
    
    def paginate(self, **kwargs):
        """
        Get paginated retention events.
        
        Args:
            **kwargs: Query parameters
        
        Returns:
            PageIterator for retention events
        """
        return self.client.paginate(self.endpoint, params=kwargs)

File: php/src/CourtListener/Api/RetentionEvents.php
<?php

namespace CourtListener\Api;

use CourtListener\CourtListenerClient;
use CourtListener\Exceptions\CourtListenerException;

/**
 * RetentionEvents API endpoint
 */
class RetentionEvents extends BaseApi
{
    protected string $endpoint = 'retention-events/';

    /**
     * Constructor
     *
     * @param CourtListenerClient $client
     */
    public function __construct(CourtListenerClient $client)
    {
        parent::__construct($client);
    }

    /**
     * List retention events with advanced filtering
     *
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function listRetentionEvents(array $params = []) {
        return $this->list($params);
    }

    /**
     * Get a specific retention events by ID
     *
     * @param int|string $id RetentionEvents ID
     * @param array $params Query parameters
     * @return array
     * @throws CourtListenerException
     */
    public function getRetentionEvents(array $params = []) {
        return $this->get($id, $params);
    }

    /**
     * Search retention events
     *
     * @param array $params Search parameters
     * @return array
     * @throws CourtListenerException
     */
    public function searchRetentionEvents(array $params = []) {
        return $this->search($params);
    }
}
